comp/languagedetection/client/client.go

```diff
@@ -0,0 +1,328 @@
+	// from the kubelet
+	processesWithoutPod          []workloadmeta.Event
+	retryProcessWithoutPodPeriod time.Duration
+
+	// periodicalFlushPeriod sets the interval between two periodical flushes
+	periodicalFlushPeriod time.Duration
+}
+
+// newClient creates a new Client
+func newClient(
+	deps dependencies,
+) Component {
+	if !deps.Config.GetBool("language_detection.enabled") || !deps.Config.GetBool("cluster_agent.enabled") {
+		return util.NewNoneOptional[Component]()
+	}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+
+	cl := &client{
+		ctx:                          ctx,
+		cancel:                       cancel,
+		logger:                       deps.Log,
+		newUpdatePeriod:              deps.Config.GetDuration("language_detection.client_period"),
+		mutex:                        sync.Mutex{},
+		telemetry:                    newComponentTelemetry(deps.Telemetry),
+		currentBatch:                 make(batch),
+		processesWithoutPod:          make([]workloadmeta.Event, 0),
+		retryProcessWithoutPodPeriod: deps.Config.GetDuration("kubelet_cache_pods_duration") * time.Second,
+		freshlyUpdatedPods:           make(map[string]struct{}),
+		periodicalFlushPeriod:        periodicalFlushPeriod,
+	}
+	deps.Lc.Append(fx.Hook{
+		OnStart: cl.start,
+		OnStop:  cl.stop,
+	})
+
+	return util.NewOptional[Component](cl)
+}
+
+// start starts streaming languages to the Cluster-Agent
+func (c *client) start(_ context.Context) error {
+	c.logger.Infof("Starting language detection client")
+	go c.run()
+	return nil
+}
+
+// stop stops the client
+func (c *client) stop(_ context.Context) error {
+	c.cancel()
+	return nil
+}
+
+// run starts processing events and starts streaming
+func (c *client) run() {
+	defer c.logger.Infof("Shutting down language detection client")
+	// workloadmeta can't be initialized in the constructor or provided as a dependency until workloadmeta is refactored as a component
+	if c.store == nil {
+		c.store = workloadmeta.GetGlobalStore() // TODO(components): should be replaced by components
+	}
+
+	eventCh := c.store.Subscribe(
+		subscriber,
+		workloadmeta.NormalPriority,
+		workloadmeta.NewFilter(
+			[]workloadmeta.Kind{
+				workloadmeta.KindKubernetesPod, // Subscribe to pod events to clean up the current batch when a pod is deleted
+				workloadmeta.KindProcess,       // Subscribe to process events to populate the current batch
+			},
+			workloadmeta.SourceAll,
+			workloadmeta.EventTypeAll,
+		),
+	)
+	defer c.store.Unsubscribe(eventCh)
+
+	metricTicker := time.NewTicker(runningMetricPeriod)
+	defer metricTicker.Stop()
+
+	retryProcessWithoutPodTicker := time.NewTicker(c.retryProcessWithoutPodPeriod)
+	defer retryProcessWithoutPodTicker.Stop()
+
+	go c.startStreaming()
+
+	for {
+		select {
+		case eventBundle := <-eventCh:
+			c.processEvent(eventBundle)
+		case <-retryProcessWithoutPodTicker.C:
+			c.retryProcessEventsWithoutPod()
+		case <-metricTicker.C:
+			c.telemetry.Running.Set(1)
```

Comment:
Why not use health check mechanism instead?

Reply:
Do we have an example ?

Reply:
Done ! :thx: